fix: stdlib `dataclass` converted into _pydantic_ `dataclass` still equals its stdlib `dataclass` equivalent